File: ejercicio5-kmp-camara-microfono/CamaraMicKMP/shared/src/commonMain/kotlin/mx/ipn/escom/camaramic/MediaItem.kt
package mx.ipn.escom.camaramic

import kotlinx.serialization.Serializable

@Serializable
enum class MediaKind { PHOTO, AUDIO }

@Serializable
data class MediaItem(
    val id: String,
    val kind: MediaKind,
    val fileName: String,
    val category: String,
    val createdAtEpochMillis: Long,
    val durationSeconds: Double = 0.0,
)

File: ejercicio3-camara-microfono-ios/CamaraMicrofonoIOS/Models/MediaItem.swift
import Foundation

enum MediaKind: String, Codable {
    case photo
    case audio
}

/// Plain struct mirror of the `CapturedItemEntity` Core Data record, used
/// throughout the UI layer so views don't depend on Core Data types directly.
struct MediaItem: Identifiable, Hashable {
    let id: UUID
    let kind: MediaKind
    let fileName: String
    let category: String
    let createdAt: Date
    let durationSeconds: Double?

    func fileURL(in directory: URL) -> URL {
        directory.appendingPathComponent(fileName)
    }
}
